### PR TITLE
data/complete: fix three out of four shellcheck warnings in data/complete

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

File: data/completion/etelpmoc.sh [hunk 1]
```diff
@@ -209,4 +213,4 @@ fi
 echo "${!_compopts[@]}"
 echo "$_bounce"
 echo ""
-printf "%s\n" "${COMPREPLY[@]}"
+printf "%s\\n" "${COMPREPLY[@]}"
```
Comment (bboozzoo) [inline data/completion/etelpmoc.sh]: or `'%s\n'`

File: data/completion/complete.sh [hunk 2]
```diff
@@ -116,7 +116,7 @@ else
     _complete_from_snap_maybe() {
         local etel=snap/core/current/usr/lib/snapd/etelpmoc.sh
         # catch /snap/bin and /var/lib/snapd/snap/bin
-        if [[ "$(which "$1")" =~ /snap/bin/ && ( -e "/var/lib/snapd/$etel" || -e "/$etel" ) ]]; then
+        if [[ "$(command -v "$1")" =~ /snap/bin/ && ( -e "/var/lib/snapd/$etel" || -e "/$etel" ) ]]; then
```
Comment (chipaca) [inline data/completion/complete.sh]: the problem with this is that if you have an *bash* alias to a command, it'll match this
```sh
~$ alias boo=/snap/bin/http
~$ command -v boo
alias boo='/snap/bin/http'
~$ [[ "$( command -v boo )" =~ /snap/bin/ ]] && echo YES
YES
```
but the completer will have no idea how to handle it.

Comment (zyga) [reply data/completion/complete.sh]: Hmm, hmm, shall I drop this patch then?

Comment (zyga) [reply data/completion/complete.sh]: Done now

File: data/completion/complete.sh [hunk 3]
```diff
@@ -116,7 +116,7 @@ else
     _complete_from_snap_maybe() {
         local etel=snap/core/current/usr/lib/snapd/etelpmoc.sh
         # catch /snap/bin and /var/lib/snapd/snap/bin
-        if [[ "$(which "$1")" =~ /snap/bin/ && ( -e "/var/lib/snapd/$etel" || -e "/$etel" ) ]]; then
+        if [[ "$(command -v "$1")" =~ ^(/var/lib/snapd)?/snap/bin/ && ( -e "/var/lib/snapd/$etel" || -e "/$etel" )  ]]
```
Comment (chipaca) [inline data/completion/complete.sh]: um, you ate the "then" at the end there

Comment (zyga) [reply data/completion/complete.sh]: Fixed, thanks!

**Pull request merged**